lib/perl/Genome/InstrumentData/Imported.pm

```diff
@@ -417,15 +417,21 @@ sub bam_path {
 
     my ($allocation) = $self->disk_allocation;
     unless ($allocation) {
-        $self->error_message("Found no disk allocation for imported instrument data " . $self->id, ", so cannot find bam!");
-        die $self->error_message;
+        $self->fatal_message("Found no disk allocation for imported instrument data %s, so cannot find bam!", $self->__display_name__);
     }
 
     my $bam_file = $allocation->absolute_path . "/all_sequences.bam";
 
-    return if not -e $bam_file;
+    return $bam_file if -e $bam_file;
+
+    my @possible_bams = glob($allocation->absolute_path . "/*.bam");
+    if (@possible_bams == 1) {
+        return $possible_bams[0];
+    } elsif (@possible_bams > 1) {
+        $self->fatal_message('Multiple BAMs found in allocation for imported instrument data: %s.  Name one "all_sequences.bam" to have it used here.', $self->__display_name__);
+    }
 
```

Comment:
What if the glob returns nothing because there is no bam file?

Reply:
We fall the to empty `return;` at the end and assume it's something else

Reply:
okay - fair enough!

Reply:
(This still doesn't handle the issue of running from somewhere where the path isn't mounted... but that isn't a new problem and it extends far beyond this one accessor.)